stdlib/public/core/StringGuts.swift

```diff
@@ -417,6 +417,89 @@ extension _StringGuts {
   }
 }
 
+#if _runtime(_ObjC)
+extension _StringGuts {
+
+  private static var associationKey: UnsafeRawPointer {
+    // We never dereference this, we only use this address as a unique key
+    unsafe unsafeBitCast(
+      ObjectIdentifier(_StringGuts.self),
```

Comment:
How fast is this metatype lookup in practice? (Elsewhere we sometimes decided to use the address of a dummy symbol instead; but this may in fact be the better pattern.)

Reply:
Oh, on second look, this isn't exactly right -- _StringGuts is a public type, so it is trivial to generate this ID outside the stdlib. It would be better to use the metatype of an internal type here.

(It's not that hard to work out what ID the stdlib uses, but making it too easy sends the wrong signal.)

Reply:
The two options are to cast about for a suitable type, or make up a private type and use that
```suggestion
      ObjectIdentifier(__StringStorage.self),
```

Reply:
`__StringStorage` is technically internal, but it is the isa of native Swift strings bridged to ObjC, so it is still very public. Something like `_StringObject` would be a better choice, I think -- a pure internal (non-`@usableFromInline`) or private type would be even better.

Reply:
I'll go with a new made-up private type, then.